pkg/cmd/pipelineresource/list.go

```diff
@@ -61,7 +63,7 @@ func listCommand(p cli.Params) *cobra.Command {
 		Args: cobra.MinimumNArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
 
-			if err := validinput.NamespaceExists(p); err != nil {
+			if err := validinput.NamespaceExists(p); err != nil && !opts.AllNamespaces {
```

Comment:
We don't even need to check whether namespace exists if `all-namespace` is true

Reply:
We need to, as the test for this needs `-n` 
This was confirmed in the previous PRs where we were adding the --all-namespaces flag.
And when we are getting the output for all namespaces we do not need to check if the particular namespace exists.

Reply:
You can do this to avoid checking for the namespace and unit tests should work: 

```go
if !opts.AllNamespaces {
    if err := validinput.NamespaceExists(p); err != nil {
        return err
    }
}

namespace := p.Namespace()
if opts.AllNamespaces {
    namespace = ""
}
```

Reply:
I have raised #974 for other commands and removed `-n` in tests also. And I have done the same as daniel mentioned above.